tests/ResponsiveProvider.test.js

```diff
@@ -152,23 +152,30 @@ describe('<ResponsiveProvider />', () => {
         expect(listenersRemovedAfterUnmount).toBe(listenersAdded);
     });
 
-    test('should not call console.log in the test environment', () => {
-        window.resizeTo(1024, 768);
+    test.each(
+        ['production', 'test'],
+        'should not call console.log in the % environment',
+        (env) => {
+            process.env.NODE_ENV = env;
+            window.resizeTo(1024, 768);
 
-        render(
-            <ResponsiveProvider {...props}>
-                <div>Test</div>
-            </ResponsiveProvider>
-        );
+            render(
+                <ResponsiveProvider {...props}>
+                    <div>Test</div>
+                </ResponsiveProvider>
+            );
 
-        expect(global.console.group).not.toHaveBeenCalled();
-        expect(global.console.log).not.toHaveBeenCalled();
-        expect(global.console.groupEnd).not.toHaveBeenCalled();
-    });
+            expect(global.console.group).not.toHaveBeenCalled();
+            expect(global.console.log).not.toHaveBeenCalled();
+            expect(global.console.groupEnd).not.toHaveBeenCalled();
+
+            // Clean the global changes
+            process.env.NODE_ENV = 'test';
+        }
+    );
 
-    test('should call console.log only outside the test environment', () => {
-        // Set this as prod environment
-        process.env.NODE_ENV = 'dev';
+    test('should call console.log only for development environment', () => {
```

Comment:
Do you really mean **only** development?  If so, then you should probably change the condition in the source code to reflect that.

Reply:
Do you prefer "not test and not production"? I mean it would more correct but less readable

Reply:
Yea it's not as pretty but you could also remove the "only".   I'm not too knowledgeable in this area but do you foresee any environment besides dev, production, and test?   And do you think people will want those logs in all of those other "unknown environments that are not test and production"?  Imagine I have an environment "xyz", but I don't want your logging, now there is no way to turn it off, and the logs aren't as "opt-in".

It seems like a better approach would be to specify exactly which environments you want it in, rather than those environments you do NOT want it in.  What do you think?

Reply:
Doesn't seem a bad idea, there must be people who have more environments, I will change it to `=== 'development`

Reply:
Changed